Remove obsolete network.txt & add examples for wpa-supplicant

foosel opened the pull request:
The `<dist>-network.txt`  mechanism doesn't work anymore under stretch. Considering that @guysoft has been pushing to obsolete it anyhow in favor of the more flexible `<dist>-wpa-supplicant.txt` approach anyhow, this PR removes the old mechanism and related files and adds an example `<dist>-wpa-supplicant.txt` (with configuration samples for WPA/WPA2, unencrypted and WEP) that gets merged with the distributed version during build.

I've tested this against a WPA2-PSK, WEP and unsecured network.

guysoft commented:
Thanks for the pull request!
Just pointing out "you missed a spot". The module config is not updated:
https://github.com/guysoft/CustomPiOS/blob/CustomPiOS/src/modules/network/config

I guess we should change ``DIST_NETWORK_FILE`` to use ``NETWORK_CONFIG_FILE`` instead, and update that. It lets other distros modify the name of the file.

foosel commented:
Updated to fix the left-over. As discussed on hangouts I removed the config options completely since I couldn't get them to work at all (they were always empty inside the chroot script -  maybe an issue with var export?) so for now the name is static.
